cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -195,16 +203,57 @@ public void onClose(KubernetesClientException e) {
         });
     }
 
+    /**
+     * Searches through the conditions in the supplied status instance and checks if the type of each condition matches
+     * one of the defined State enum values. This method will return the first condition instance whose type which
+     * matches one of the State enum values.
+     *
+     * @param status The status instance whose conditions will be searched.
+     * @return The first condition which matches one of the State enum values. If there is no matching type then
+     *         this method will return null.
+     */
+    private Condition getCCStatusCondition(KafkaRebalanceStatus status) {
+        if (status.getConditions() != null) {
+            for (Condition condition : status.getConditions()) {
+                if (condition.getType() != null) {
+                    String conditionType = condition.getType();
+                    for (State state : State.values()) {
+                        if (conditionType.equals(state.toString())) {
+                            return condition;
+                        }
```

Comment:
do you think we can have something using the "stream" API?

Reply:
So something like:
```
if (condition.getType() != null) {
    String conditionType = condition.getType();
    if (Stream.of(State.values()).anyMatch(state -> conditionType.equals(state.toString()))) {
        return condition;
    }
}
```